File: js/src/instructions/updateInstruction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.updateInstruction = void 0;
const buffer_1 = require("buffer");
const web3_js_1 = require("@solana/web3.js");
const int_1 = require("../int");
function updateInstruction(nameProgramId, nameAccountKey, offset, input_data, nameUpdateSigner) {
    const buffers = [
        buffer_1.Buffer.from(Int8Array.from([1])),
        offset.toBuffer(),
        new int_1.Numberu32(input_data.length).toBuffer(),
        input_data,
    ];
    const data = buffer_1.Buffer.concat(buffers);
    const keys = [
        {
            pubkey: nameAccountKey,
            isSigner: false,
            isWritable: true,
        },
        {
            pubkey: nameUpdateSigner,
            isSigner: true,
            isWritable: false,
        },
    ];
    return new web3_js_1.TransactionInstruction({
        keys,
        programId: nameProgramId,
        data,
    });
}
exports.updateInstruction = updateInstruction;

File: python/src/instructions/update_instruction.py
from solders.instruction import Instruction, AccountMeta
from solders.pubkey import Pubkey

from int import Numberu32


def update_instruction(
        name_program_id: Pubkey,
        name_account: Pubkey,
        offset: Numberu32,
        input_data: bytes,
        name_update_signer: Pubkey,
) -> Instruction:
    buffers = [
        bytes([1]),
        offset.to_bytes(),
        Numberu32(len(input_data)).to_bytes(),
        input_data,
    ]
    data = b"".join([bytes(buffer) for buffer in buffers])
    accounts = [
        AccountMeta(
            pubkey=name_account,
            is_signer=False,
            is_writable=True,
        ),
        AccountMeta(
            pubkey=name_update_signer,
            is_signer=True,
            is_writable=False,
        ),
    ]

    return Instruction(
        program_id=name_program_id,
        accounts=accounts,
        data=data,
    )